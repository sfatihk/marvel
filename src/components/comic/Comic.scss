@import "../../assets/style/mixin.scss";
@import "../../assets/style/colors.scss";

.comic {
  margin: 1vmin;
  position: relative;
  width: 100%;
  height: 400px;
  display: flex;
  flex-direction: row;
  cursor: pointer;

  @media (max-width: 768px) {
    flex-direction: column;
    margin-bottom: 30px;
    height: auto;
  }
}
.comic:hover {
  background-color: $color7;
  border-radius: 10px;
}

.comic img {
  transform-origin: 100 100;
  transition: transform 0.5s ease;
  flex-shrink: 0;
  min-height: 100%;

  max-height: 100%;
  @include characterCoverFrame;
  @media (max-width: 768px) {
    flex-direction: column;
    max-height: unset;
    max-width: 100%;
  }
}

.comic__content {
  flex: 1;
  padding-left: 10px;
  @media (max-width: 768px) {
    padding-left: 0;
  }
}

.comic__title {
  text-decoration: none;
  color: $color1;
  font-size: 24px;
  @media (max-width: 768px) {
    font-size: 40px;
  }
}

.comic__pageCount {
  font-size: 18px;
  letter-spacing: 2px;
  color: $color6;
  @media (max-width: 768px) {
    font-size: 22px;
  }
}

.comic__publishDate {
  font-size: 18px;
  letter-spacing: 2px;
  color: $color6;
  @media (max-width: 768px) {
    font-size: 22px;
  }
}

.comic__price {
  font-size: 18px;
  letter-spacing: 2px;
  color: $color6;
  @media (max-width: 768px) {
    font-size: 22px;
  }
}
