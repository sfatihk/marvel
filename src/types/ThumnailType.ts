export type ThumnailType = {
  extension: string;
  path: string;
};
