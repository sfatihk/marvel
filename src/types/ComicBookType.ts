import { ThumnailType } from "./ThumnailType";

type UrlType = {
  type: "detail" | "purchase" | "reader" | "inAppLink";
  url: string;
};
type DateType = {
  type: "onsaleDate" | "focDate" | "unlimitedDate" | "digitalPurchaseDate";
  date: string;
};
type PriceType = {
  type: "printPrice" | "digitalPurchasePrice";
  price: number;
};
export type ComicBookType = {
  name: string;
};

export type ExtendedComicBookType = {
  id: number;
  title: string;
  pageCount: number;
  urls: UrlType[];
  dates: DateType[];
  prices: PriceType[];
  thumbnail: ThumnailType;
};
