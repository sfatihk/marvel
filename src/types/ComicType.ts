import { ComicBookType } from "./ComicBookType";
export type ComicType = {
  items: ComicBookType[];
};
