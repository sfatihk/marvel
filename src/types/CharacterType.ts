import { ThumnailType } from "./ThumnailType";
import { ComicType } from "./ComicType";

export type CharacterType = {
  id: number;
  name: string;
  thumbnail: ThumnailType;
  description: string;
  comics: ComicType;
};
