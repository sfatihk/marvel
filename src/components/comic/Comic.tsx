import React, { useState } from "react";
import { ExtendedComicBookType } from "../../types/ComicBookType";

//styles
import "./Comic.scss";
const Comic = ({
  id,
  title,
  pageCount,
  dates,
  prices,
  thumbnail,
  urls,
}: ExtendedComicBookType) => {
  const [publishedDate] = useState(
    dates.find((date) => date.type === "onsaleDate")?.date
  );
  return (
    <a
      className="comic"
      href={urls.find((url) => url.type === "detail")?.url}
      target="_blank"
      rel="noreferrer"
    >
      <img
        src={thumbnail.path + "." + thumbnail.extension}
        alt=""
        className="comic__thumbnail"
      />
      <div className="comic__content">
        <div className="comic__title">{title}</div>
        <div className="comic__pageCount">Page count : {pageCount || "-"}</div>
        <div className="comic__publishDate">
          Published :{" "}
          {publishedDate ? new Date(publishedDate).toLocaleDateString() : "-"}
        </div>
        <div className="comic__price">
          Price :{" "}
          {"$" + prices.find((price) => price.type === "printPrice")?.price ||
            "-"}
        </div>
      </div>
    </a>
  );
};

export default Comic;
