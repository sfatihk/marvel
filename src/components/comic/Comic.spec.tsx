import renderer from "react-test-renderer";
import { ExtendedComicBookType } from "../../types/ComicBookType";

import Comic from "./Comic";

describe("Comic components", () => {
  const initialState: ExtendedComicBookType = {
    id: 123,
    title: "Comic Book",
    dates: [{ type: "onsaleDate", date: "2020-03-18T00:00:00-0400" }],
    pageCount: 123,
    prices: [{ type: "printPrice", price: "321" }],
    thumbnail: { extension: "png", path: "xyz" },
    urls: [{ type: "detail", url: "https://google.com" }],
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Comic {...initialState} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
