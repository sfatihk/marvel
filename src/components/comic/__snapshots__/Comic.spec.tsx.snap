// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comic components renders correctly 1`] = `
<div
  className="comic"
>
  <img
    alt=""
    className="comic__thumbnail"
    src="xyz.png"
  />
  <div
    className="comic__content"
  >
    <div
      className="comic__title"
    >
      Comic Book
    </div>
    <div
      className="comic__pageCount"
    >
      Page count : 
      123
    </div>
    <div
      className="comic__publishDate"
    >
      Published :
       
      18.03.2020
    </div>
    <div
      className="comic__price"
    >
      Price :
       
      $321
    </div>
    <a
      className="comic__link"
      href="https://google.com"
      rel="noreferrer"
      target="_blank"
    >
      details...
    </a>
  </div>
</div>
`;
